In [15]:
import pandas as pd
from sklearn.model_selection import train_test_split
file_path = ("/Users/manishashah/Desktop/UniversalBank_Ensemble(1).csv")
data = pd.read_csv(file_path)

# Display the first few rows of the dataset
data.head()

Age  Experience  Income  Family  Education  Mortgage  Personal Loan  \
0   25           1      49       4          1         0              0   
1   45          19      34       3          1         0              0   
2   39          15      11       1          1         0              0   
3   35           9     100       1          2         0              0   
4   35           8      45       4          2         0              0   

   Securities Account  CreditCard  
0                   1           0  
1                   1           0  
2                   0           0  
3                   0           0  
4                   0           1

In [16]:
data = pd.read_csv(file_path)

# Split the data into features and target
X = data.drop('Personal Loan', axis=1)
y = data['Personal Loan']

# Split the dataset into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [17]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import BaggingClassifier, RandomForestClassifier, GradientBoostingClassifier
from sklearn.metrics import accuracy_score, classification_report

# Initialize the models
decision_tree = DecisionTreeClassifier(random_state=42)
bagging = BaggingClassifier(base_estimator=DecisionTreeClassifier(), n_estimators=100, random_state=42)
random_forest = RandomForestClassifier(n_estimators=100, random_state=42)
boosting = GradientBoostingClassifier(n_estimators=100, random_state=42)


In [18]:
# Train the models
decision_tree.fit(X_train, y_train)
bagging.fit(X_train, y_train)
random_forest.fit(X_train, y_train)
boosting.fit(X_train, y_train)

/Users/manishashah/anaconda3/envs/tensor/lib/python3.10/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(


GradientBoostingClassifier(random_state=42)

In [19]:
from sklearn.ensemble import GradientBoostingClassifier

# Initialize the Gradient Boosting Classifier
gb_classifier = GradientBoostingClassifier(random_state=42)

# Train the model using the training data
gb_classifier.fit(X_train, y_train)

# Make predictions using the test data
predictions = gb_classifier.predict(X_test)

# Evaluate the model (example: using accuracy)
from sklearn.metrics import accuracy_score
accuracy = accuracy_score(y_test, predictions)
accuracy

0.9813333333333333

The models have been trained, and their accuracy on the test set is as follows:

Decision Tree Accuracy: 96.27%
Bagging Accuracy: 98.07%
Random Forest Accuracy: 98.13%
Boosting Accuracy: 98.13%


In [20]:

# Predict on the test set
dt_predictions = decision_tree.predict(X_test)
bagging_predictions = bagging.predict(X_test)
rf_predictions = random_forest.predict(X_test)
boosting_predictions = boosting.predict(X_test)

# Calculate accuracies
dt_accuracy = accuracy_score(y_test, dt_predictions)
bagging_accuracy = accuracy_score(y_test, bagging_predictions)
rf_accuracy = accuracy_score(y_test, rf_predictions)
boosting_accuracy = accuracy_score(y_test, boosting_predictions)

(dt_accuracy, bagging_accuracy, rf_accuracy, boosting_accuracy)


(0.9626666666666667,
 0.9806666666666667,
 0.9813333333333333,
 0.9813333333333333)

In [21]:
# Caroline's details
caroline_details = {
    'Age': 32,
    'Experience': 6,
    'Income': 200,  
    'Family': 3,    
    'Education': 3, 
    'Mortgage': 0,  
    'Securities Account': 1, 
    'CreditCard': 1 
}

# Convert to DataFrame for prediction
caroline_df = pd.DataFrame([caroline_details])

# Predictions for Caroline
dt_prediction_caroline = decision_tree.predict(caroline_df)[0]
bagging_prediction_caroline = bagging.predict(caroline_df)[0]
rf_prediction_caroline = random_forest.predict(caroline_df)[0]
boosting_prediction_caroline = boosting.predict(caroline_df)[0]

(dt_prediction_caroline, bagging_prediction_caroline, rf_prediction_caroline, boosting_prediction_caroline)


(1, 1, 1, 1)